sklearn/decomposition/_fastica.py

```diff
@@ -584,15 +584,16 @@ def transform(self, X, copy=True):
             and n_features is the number of features.
 
         copy : bool, default=True
-            If False, data passed to fit are overwritten. Defaults to True.
+            If False, data passed to fit can be overwritten. Defaults to True.
 
         Returns
         -------
         X_new : ndarray of shape (n_samples, n_components)
         """
         check_is_fitted(self)
 
-        X = check_array(X, copy=copy, dtype=FLOAT_DTYPES)
+        X = self._validate_data(X, copy=(copy and self.whiten),
```

Comment:
Isn't this a change of behavior?

Reply:
This is a small memory efficiency optim :)